db/attachment.go

```diff
@@ -105,6 +105,9 @@ func (db *Database) storeAttachments(doc *document, body Body, generation int, p
 			if revpos, ok := base.ToInt64(meta["revpos"]); !ok || revpos < 1 {
 				return nil, base.HTTPErrorf(400, "Missing/invalid revpos in stub attachment %q", name)
 			}
+			if meta["digest"] == nil {
+				return nil, base.HTTPErrorf(400, "Missing digest in stub attachment %q", name)
+			}
```

Comment:
I don't this change is needed - as in the previous code, we only want to return an error on an empty digest if the attempted parent lookup fails.